src/audio/dai.c

```diff
@@ -1161,7 +1161,13 @@ static int dai_ts_config(struct comp_dev *dev)
 	cfg->direction = dai->direction;
 	cfg->index = dd->dai->index;
 	cfg->dma_id = dd->dma->plat_data.id;
-	cfg->dma_chan_index = dd->chan->index;
+	if (dd->chan) {
```

Comment:
Can we add a comment in the dai API header doxygen comments that state when this API can be called.

Reply:
@singalsu I would move the dd->chan check just at the top of the function and leave this line untouched.

E. g

```
if (!dd->chan) {
      comp_err()...
     return -EINVAL;
}

```

Reply:
Fwiw, we seem to be lacking some general documentation in the DAI logic but AFAICT this fix should do the trick.

Reply:
Yep, my add seemed to be only one there. 

I'll test updated here and convert to non-draft if OK.